packages/react-native-gesture-handler/src/web/handlers/GestureHandler.ts

```diff
@@ -356,21 +356,21 @@ export default abstract class GestureHandler implements IGestureHandler {
       return;
     }
 
-    const { onGestureHandlerEvent, onGestureHandlerTouchEvent }: PropsRef = this
-      .propsRef.current as PropsRef;
+    const { onGestureHandlerEvent, onGestureHandlerTouchEvent }: PropsRef =
+      this.propsRef.current;
 
     const touchEvent: ResultTouchEvent | undefined =
       this.transformTouchEvent(event);
 
     if (touchEvent) {
       if (
-        this.actionType === ActionType.NATIVE_DETECTOR ||
-        this.actionType === ActionType.NATIVE_DETECTOR_ANIMATED_EVENT
+        onGestureHandlerTouchEvent &&
+        (this.actionType === ActionType.NATIVE_DETECTOR ||
+          this.actionType === ActionType.NATIVE_DETECTOR_ANIMATED_EVENT)
       ) {
         invokeNullableMethod(onGestureHandlerTouchEvent, touchEvent);
-      } else {
-        invokeNullableMethod(onGestureHandlerEvent, touchEvent);
       }
+      invokeNullableMethod(onGestureHandlerEvent, touchEvent);
```

Comment:
```suggestion
      } else {
        invokeNullableMethod(onGestureHandlerEvent, touchEvent);
      }
```

With touch events, we want to keep them on separate paths 😅